### PR TITLE
Add Simon's constructor for elliptic curves.

### PR DESCRIPTION
**Pull request by HechtiDerLachs**
CC: @simonbrandhorst

### CODE REVIEW

File: test/EllCrv/EllCrv.jl [hunk 1]
```diff
@@ -291,3 +291,15 @@
     @test_throws ErrorException P1//5
   end
 end
+
+@testset "additional constructors" begin
+  R, (x, y) = polynomial_ring(QQ, [:x, :y])
+  f = y^2 - x^3 - 3*x^2 + 7*x - 4
+  Hecke.elliptic_curve(f, x, y)
```
Comment (thofma) [inline test/EllCrv/EllCrv.jl:298]:
```suggestion
elliptic_curve(f, x, y)
```

File: test/EllCrv/EllCrv.jl [hunk 2]
```diff
@@ -291,3 +291,15 @@
...
+
+  pt, t = polynomial_ring(QQ, :t)
+  kt = fraction_field(pt)
+  R, (x, y) = polynomial_ring(kt, [:x, :y])
+  f = y^2 - x^3 - 3*t*x^2 + 7*x - 4*t^2 - 3
+  Hecke.elliptic_curve(f, x, y)
```
Comment (thofma) [inline test/EllCrv/EllCrv.jl:304]:
```suggestion
  elliptic_curve(f, x, y)
```

**Comment by thofma** [comment]
A docstring would be nice

**Comment by HechtiDerLachs** [comment]
> A docstring would be nice

Really? I was afraid, this would not be appreciated because of too many documented methods of the same function. But I can try to add it. 

Btw: It seems you already had similar functionality. Is that true?

**Comment by thofma** [comment]
At the moment there are two methods documented, so adding another won't hurt. And I don't think we have anything that accepts bivariate polynomials.

**Pull request merged**